gdctools/lib/common.py

```diff
@@ -133,6 +133,24 @@ def map_blank_to_na(csvfile):
     for row in csvfile:
         yield map(lambda f: f if f != '' else 'NA', row)
 
+def rearrange_columns(csvfile, col_scramble=None):
+    """
+    col_scramble is a list whose index is for the column in the output file and whose values 
+    represent column numbers of the input file.   Eg 0,2,3,4,1 will move column 1 of the input file to be after the others.  If col_scramble is None, column order is left unchanged.
+    """
+    if col_scramble:
+        len_expected = len(col_scramble)
```

Comment:
Shouldn't this be `len_expected = max(col_reorder) + 1`? This adds robustness to the function, and enables the choice to drop arbitrary columns.